src/main/java/io/github/rabobank/shadow_tool/ShadowFlow.java

```diff
@@ -247,14 +248,38 @@ public ShadowFlowBuilder<T> withExecutorService(final ExecutorService executorSe
 
         /**
          * This configures the shadow flow to log the values of the differences found between the two flows.
-         * Since the data is potentially sensitive, encryption is required.
+         * Since the data is potentially sensitive, encryption is required. Mutually exclusive with
+         * {@link ShadowFlowBuilder#withCipher(Cipher cipher) withCipher }. This method will use a cipher with
+         * RSA/ECB/OAEPWITHSHA-256ANDMGF1PADDING
          *
          * @param publicKey The public RSA key used for encryption, should be at least 2048 bits.
          * @return This builder.
          */
         public ShadowFlowBuilder<T> withEncryption(final PublicKey publicKey) {
             try {
-                encryptionService = new EncryptionService(publicKey);
+                final String ALGORITHM = "RSA";
+                final String ALGORITHM_MODE_PADDING = ALGORITHM + "/ECB/OAEPWITHSHA-256ANDMGF1PADDING";
```

Comment:
Please move these to "constants" at the top of the class (`private static final`)

Reply:
Happy to change it if you want. But will it ever be used outside of this method? Because this is pretty specific for the creation of this cipher

Reply:
Well, it would be (very) slightly better, because they would be static. On the other hand it only happens in the builder. It's fine like this!